packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -49,44 +57,86 @@ contract AnchorStateRegistry is Initializable, ISemver {
+
+    /// @param _disputeGameFinalityDelaySeconds The dispute game finality delay in seconds.
+    constructor(uint256 _disputeGameFinalityDelaySeconds) {
+        DISPUTE_GAME_FINALITY_DELAY_SECONDS = _disputeGameFinalityDelaySeconds;
         _disableInitializers();
     }
 
     /// @notice Initializes the contract.
     /// @param _superchainConfig The address of the SuperchainConfig contract.
     /// @param _disputeGameFactory The address of the DisputeGameFactory contract.
-    /// @param _portal The address of the OptimismPortal contract.
     /// @param _startingAnchorRoot The starting anchor root.
     function initialize(
         ISuperchainConfig _superchainConfig,
         IDisputeGameFactory _disputeGameFactory,
-        IOptimismPortal2 _portal,
-        OutputRoot memory _startingAnchorRoot
+        OutputRoot memory _startingAnchorRoot,
+        GameType _startingRespectedGameType
     )
         external
         initializer
     {
         superchainConfig = _superchainConfig;
         disputeGameFactory = _disputeGameFactory;
-        portal = _portal;
         startingAnchorRoot = _startingAnchorRoot;
+        respectedGameType = _startingRespectedGameType;
+        retirementTimestamp = uint64(block.timestamp);
+    }
+
+    /// @notice Returns whether the contract is paused.
+    /// @return Whether the contract is paused.
+    function paused() public view returns (bool) {
+        return superchainConfig.paused();
     }
 
-    /// @notice Returns the respected game type.
-    /// @return The respected game type.
-    function respectedGameType() public view returns (GameType) {
-        return portal.respectedGameType();
+    /// @notice Returns the dispute game finality delay in seconds.
+    /// @return The dispute game finality delay in seconds.
+    function disputeGameFinalityDelaySeconds() external view returns (uint256) {
+        return DISPUTE_GAME_FINALITY_DELAY_SECONDS;
+    }
+
+    /// @notice Allows the Guardian to set the respected game type.
+    /// @param _gameType The new respected game type.
+    function setRespectedGameType(GameType _gameType) external {
+        if (msg.sender != superchainConfig.guardian()) revert AnchorStateRegistry_Unauthorized();
+        respectedGameType = _gameType;
+        emit RespectedGameTypeSet(_gameType);
+    }
+
+    /// @notice Allows the Guardian to update the retirement timestamp.
+    function updateRetirementTimestamp() external {
+        if (msg.sender != superchainConfig.guardian()) revert AnchorStateRegistry_Unauthorized();
+        retirementTimestamp = uint64(block.timestamp);
+        emit RetirementTimestampSet(block.timestamp);
+    }
+
+    /// @notice Allows the Guardian to blacklist a dispute game.
+    /// @param _disputeGame Dispute game to blacklist.
+    function blacklistDisputeGame(IDisputeGame _disputeGame) external {
+        if (msg.sender != superchainConfig.guardian()) revert AnchorStateRegistry_Unauthorized();
```

Comment:
this might be cleaner as a modifier, but not strong preference here

Reply:
We generally have a preference for not using modifiers because they obscure the logic a bit, will leave as-is